e3db/src/main/java/com/tozny/e3db/Client.java

```diff
@@ -989,15 +1001,39 @@ private EAKEntry getEAK(UUID writerId, UUID userId, UUID readerId, String type)
       } else {
         JsonNode eakResponse = mapper.readTree(response.body().string());
         String eak = eakResponse.get("eak").asText();
-        String publicKey = eakResponse.get("authorizer_public_key").get("curve25519").asText();
+        final String publicKey;
+        {
+          switch(Platform.crypto.suite().getEncryptionKey()) {
+            case Curve25519:
+              publicKey = eakResponse.get("authorizer_public_key").get("curve25519").asText();
+              break;
+            case P384:
+              publicKey = eakResponse.get("authorizer_public_key").get("p384").asText();
+              break;
+            default:
+              throw new IllegalStateException("Encryption key type " + Platform.crypto.suite().getEncryptionKey() + " not supported.");
+          }
+        }
```

Comment:
Could be good to keep this DRY.

Reply:
I agree but I couldnt think of a way that didn't seem brittle.